### PR TITLE
Greatly improve plugin list, including security-related enhancements

### PR DESCRIPTION
**Pull request by Gitoffthelawn**
Fix many errors.
Convert links to https when possible.
Remove link to possible malware site.
Fix broken links.
Add missing homepages.
Fix erroneous homepages.
Fix syntax errors
Add missing author credits.
Simplify descriptions and improve consistency.

### CODE REVIEW

File: src/pl.x86.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
 			"version": "2.0.0.0",
 			"id": "30d7324fd390fa336dc9ec5a249d69c825b5e59c1784097a6fdc291f9a332187",
 			"repository": "https://github.com/oleg-shilo/scripts.npp/releases/download/v2.0.0.0/NppScripts.x86.zip",
-			"description": "Notepad++ plugin for C# based automation. The plugin allows implementing a simple macros-like automations as well as the full scale script based plugins by means of C# scripts. The solution is based on CS-Script C# script engine.",
+			"description": "C# based automation.  Allows implementing a simple macros-like automations as well as the full scale script based plugins by means of C# scripts. The solution is based on CS-Script C# script engine.",
```
Comment (MAPJe71) [inline src/pl.x86.json]: A superfluous space in front of `Allows`.

Comment (Gitoffthelawn) [reply src/pl.x86.json]: Space removed.

**Comment by Gitoffthelawn** [comment]
> You forgot to update the `pl.x64.json` file.

Not forgotten.  I do not have time to work on it as well.  It's for you or others to update that file with the changes made to the x86 file.

**Comment by chcg** [comment]
@Gitoffthelawn , see failing appveyor check for valid json:
src/pl.x86.json - Expecting ':' delimiter: line 1305 column 3 (char 77170)

What I like:

> Fix many errors.
> Convert links to https when possible.
> Remove link to possible malware site.
> Fix broken links.
> Add missing homepages.
> Fix erroneous homepages.
> Fix syntax errors

, but I dislike, mixing all into one commit.

I'm still uncertain on:
Add missing author credits.
, because most likely the authors didn't add their name at the previous corresponding pluginmanager database for some reason.

I don't like the manipulations on the descriptions apart from formatting stuff like spaces, EOL, uppercase/lowercase as that was mostly written be the original authors of the plugins and I think that shouldn't be modified without their ok (at least from the ones still actively working on their plugins).

**Comment by Gitoffthelawn** [comment]
> @Gitoffthelawn , see failing appveyor check for valid json:
> src/pl.x86.json - Expecting ':' delimiter: line 1305 column 3 (char 77170)

I don't see an error.  Do you?  I'm happy to fix it if you see it.

> What I like:
> 
> > Fix many errors.
> > Convert links to https when possible.
> > Remove link to possible malware site.
> > Fix broken links.
> > Add missing homepages.
> > Fix erroneous homepages.
> > Fix syntax errors
> 
> , but I dislike, mixing all into one commit.

You're welcome.  I understand your concern, but if I made 7 commits, then people would complain about that.  I spent several hours on this as a favor, and I don't have more time to spend.

> I'm still uncertain on:
> Add missing author credits.
> , because most likely the authors didn't add their name at the previous corresponding pluginmanager database for some reason.

It doesn't matter to me.  All I did was go to the public source code pages and copy pasta'd their names.  I'm all for respecting people's privacy, but people intentionally included their names on those public pages or in their code, so I figured it was most likely accidental omissions. 

> I don't like the manipulations on the descriptions apart from formatting stuff like spaces, EOL, uppercase/lowercase as that was mostly written be the original authors of the plugins and I think that shouldn't be modified without their ok (at least from the ones still actively working on their plugins).

If you have the time, you can ask each one for approval.  Many of the descriptions were in broken English and a couple were almost unintelligible.  I didn't change the meaning of any of the descriptions; I fixed errors, made them easier to comprehend, and made them easier to translate to other languages. :)

File: src/pl.x86.json [hunk 2]
```diff
@@ -1299,18 +1299,18 @@
...
 			"repository": "https://downloads.sourceforge.net/project/npp-plugins/TextFX/TextFX%20v0.26/TextFX.v0.26.unicode.bin.zip",
-			"description": "TextFX includes numerous features to transform selected text.  Featuring: * Interactive Brace Matching * Quote handling * Character case alternation * Text rewrap * Column Lineup * Fill Text Down * Insert counter text down * Text to code conversion * Numeric Conversion * URI &amp; HTML encoding * HTML to text conversion * Submit text to W3C * Text sorting * Ascii Chart * Leading whitespace repair * Autoclose HTML &amp; braces",
-			"author": "",
-			"homepage": "http://textfx.no-ip.com/textfx/"
+			"description": "Numerous functions to transform selected text. Features: * Interactive Brace Matching * Quote handling * Character case alternation * Text rewrap * Column Lineup * Fill Text Down * Insert counter text down * Text to code conversion * Numeric Conversion * URI &amp; HTML encoding * HTML to text conversion * Submit text to W3C * Text sorting * Ascii Chart * Leading whitespace repair * Autoclose HTML &amp; braces",
+			"author": "Chris Severance",
+			"https://sourceforge.net/projects/npp-plugins/files/TextFX/"
```
Comment (chcg) [inline src/pl.x86.json]: @Gitoffthelawn The complain of appveyor is here. The homepage tag was removed.

Comment (Gitoffthelawn) [reply src/pl.x86.json]: Thank you!  Fixed!

**Comment by chcg** [comment]
@Gitoffthelawn Thanks for the contribution. After thinking again about it I came to conclusion that the active maintainers could change it back if they don't like it. I will take care on the merge with the x64 version.

**Comment by Gitoffthelawn** [comment]
You're welcome!  :)

And thank you for the merge with the x64 version.

**Pull request merged**